include/links-directory.php

```diff
@@ -285,4 +268,21 @@ public function rewrite_rules( $rules ) {
 
 		return $newrules;
 	}
+
+	/**
+	 * Removes hooks, called when switching blog @see {PLL_Base::switch_blog()}.
+	 *
+	 * @since 3.5
+	 *
+	 * @return void
+	 */
+	public function remove_hooks() {
+		remove_filter( 'language_rewrite_rules', '__return_empty_array' ); // Suppress the rules created by WordPress for our taxonomy.
+
+		foreach ( $this->get_rewrite_rules_filters() as $type ) {
+			remove_filter( $type . '_rewrite_rules', array( $this, 'rewrite_rules' ) );
+		}
+
+		remove_filter( 'rewrite_rules_array', array( $this, 'rewrite_rules' ) ); // Needed for post type archives.
```

Comment:
To avoid duplicate code, what do you think to move `get_rewrite_rules_filters()` to this class (as it's not used anywhere else than here) and rewrite it to include all filters like this?
```php
	public function get_rewrite_rules_filters() {
		// Make sure that we have the right post types and taxonomies.
		$types = array_values( array_merge( $this->model->get_translated_post_types(), $this->model->get_translated_taxonomies(), $this->model->get_filtered_taxonomies() ) );
		$types = array_merge( $this->always_rewrite, $types );

		/**
		 * Filters the list of rewrite rules filters to be used by Polylang.
		 *
		 * @since 0.8.1
		 *
		 * @param array $types The list of filters (without '_rewrite_rules' at the end).
		 */
		$types = apply_filters( 'pll_rewrite_rules', $types );
		
		filters = array(
			'language_rewrite_rules' => '__return_empty_array',
			rewrite_rules_array'     => array( $this, 'rewrite_rules' ),
		);
		
		foreach ( $types as $type ) {
			$filters[ $type . '_rewrite_rules' ] = array( $this, 'rewrite_rules' );
		}
		
		return $filters;
	}
	
	
	public function prepare_rewrite_rules() {
		if ( ! $this->model->has_languages() || ! did_action( 'wp_loaded' ) || has_filter( 'language_rewrite_rules', '__return_empty_array' ) {
			return;
		}
		
		foreach ( $this->get_rewrite_rules_filters() as $rule => $callback ) {
			add_filter( $rule, $callback );
		}
	}
	
	public function remove_hooks() {
		foreach ( $this->get_rewrite_rules_filters() as $rule => $callback ) {
			remove_filter( $rule, $callback );
		}	
	}
```

Reply:
Good idea, done in 43ab719d.